pkgs/development/libraries/openbabel/default.nix

```
@@ -1,34 +1,63 @@
-{stdenv, fetchurl, fetchpatch, cmake, zlib, libxml2, eigen, python, cairo, pcre, pkgconfig }:
+{stdenv, fetchurl, fetchpatch, cmake, zlib, libxml2, eigen, python, cairo, pcre, pkg-config, swig, rapidjson }:
 
 stdenv.mkDerivation rec {
   pname = "openbabel";
-  version = "2.4.1";
+  version = "3.1.1";
 
   src = fetchurl {
     url = "https://github.com/openbabel/openbabel/archive/openbabel-${stdenv.lib.replaceStrings ["."] ["-"] version}.tar.gz";
-    sha256 = "0xm7y859ivq2cp0q08mwshfxm0jq31xkyr4x8s0j6l7khf57yk2r";
+    sha256 = "c97023ac6300d26176c97d4ef39957f06e68848d64f1a04b0b284ccff2744f02";
   };
 
-  patches = [
-    # ARM / AArch64 fixes.
-    (fetchpatch {
-      url = "https://github.com/openbabel/openbabel/commit/ee11c98a655296550710db1207b294f00e168216.patch";
-      sha256 = "0wjqjrkr4pfirzzicdvlyr591vppydk572ix28jd2sagnfnf566g";
-    })
-  ];
 
-  # TODO : perl & python bindings;
-  # TODO : wxGTK: I have no time to compile
-  # TODO : separate lib and apps
-  buildInputs = [ zlib libxml2 eigen python cairo pcre ];
+  buildInputs = [zlib libxml2 eigen python cairo pcre swig rapidjson];
 
-  nativeBuildInputs = [ cmake pkgconfig ];
+  nativeBuildInputs = [cmake pkg-config];
+
+  pythonMajorMinor = "${python.sourceVersion.major}.${python.sourceVersion.minor}";
+
+  setupFile =
+    ''
+    from distutils.core import setup
+
+    setup(
+        name = 'pyopenbabel',
+        version = '3.1.1',
+        packages = ['openbabel'],
+        package_data = {'openbabel' : ['_openbabel.so']}
+        )
+    '';
+
+
+  buildPhase =
+    ''
+    mkdir $out
+    mkdir ob-build
+    cd ob-build
+    cmake -DRUN_SWIG=ON -DPYTHON_BINDINGS=ON -DCMAKE_INSTALL_PREFIX=$out -DLIB_INSTALL_DIR=$out  ../
```

Please do not overwrite buildPhase here. You can se the cmake flags via cmakeFlags:

```suggestion
    cmakeFlags = [  "-DRUN_SWIG=ON" "-DPYTHON_BINDINGS=ON" ];
```